test/qtiCommonRenderer/interactions/textEntry/test.js

```diff
@@ -162,6 +186,53 @@ define([
             .render($container);
     });
 
+    QUnit.cases
+        .init([
+            {
+                title: 'TextEntryInteraction contains inputmode = text',
+                item: textEntryInputmodeTextData,
+                expected: 'text'
+            },
+            {
+                title: 'TextEntryInteraction contains inputmode = numerical',
+                item: textEntryInputmodeIntegerData,
+                expected: 'numeric'
+            },
+            {
+                title: 'TextEntryInteraction contains inputmode = decimal',
+                item: textEntryInputmodeFloatData,
+                expected: 'decimal'
+            }
+        ])
+        .test('Pattern constraint - inputmode', function(data, assert) {
+            const ready = assert.async();
+            const item = _.cloneDeep(data.item);
+
+            const $container = $('#pattern-constraint-inputmode');
+
+            assert.equal($container.length, 1, 'the item container exists');
+            assert.equal($container.children().length, 0, 'the container has no children');
+            renderMatchInteraction($container, item)
+                .then(() => {
+                    const $input = $container.find('.qti-interaction.qti-textEntryInteraction');
+
+                    console.log($input.attr('inputmode'), $input);
+
+                    assert.equal(
+                        $input.attr('inputmode'),
+                        data.expected,
+                        data.title
+                    );
+                })
+                .catch(err => {
+                    assert.pushResult({
+                        result: false,
+                        message: err
+                    });
+                })
+                .then(ready);
```

Comment:
You need to tear down the item runner, otherwise the container will still contain the previous results, and this will fail your test